java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -4639,6 +4639,12 @@ static native long makeCudfColumnView(int type, int scale, long data, long dataS
 
   static native long applyBooleanMask(long arrayColumnView, long booleanMaskHandle) throws CudfException;
 
+  static native boolean hasNonEmptyNulls(long handle) throws CudfException;
+
+  static native boolean mayHaveNonEmptyNulls(long handle) throws CudfException;
```

Comment:
This API is redundant. From my experience, using it doesn't give any benefit but only performance degradation. Just remove it completely. I'll also do that on the cudf side.

Reply:
I agree as I don't understand the rationale behind adding this API. @mythrocks I think you added this, can you comment here as to whether we should even keep this API?

Reply:
Upon further reading into [this](https://github.com/rapidsai/cudf/commit/84f88ceb18225850835a9912a18e4c82245d5620), I think it's a good fast check as it only checks if the hierarchy has any nulls in it, which means it will just go over the validity buffer. It's essentially a glorified `hasNulls()` check. I think, it should stay otherwise we will have to do the expensive check all the time.

Reply:
Yes it is a fast check, but it also returns false positive result: 
```
if ((type == type_id::STRING || type == type_id::LIST) && input.has_nulls()) { return true; }
```

As such, `purge_nonempty_nulls` will be called (which is expensive) even if the input doesn't have non-empty nulls. Using it will make things slow down instead of faster. This was proved by a couple of recent cudf issues.

Reply:
Thank you for your feedback. 

Due to my lack of knowledge of this particular feature and the fact that I am leaning towards keeping this in, I would like to hear from others about their opinion. @mythrocks @hyperbolic2346

Reply:
> This was proved by a couple of recent cudf issues.
@ttnghia: Could you point me to the issues, for reference?

I'm trying to determine if this function is safe to remove. I am yet to be convinced.

> As such, purge_nonempty_nulls will be called (which is expensive) even if the input doesn't have non-empty nulls. Using it will make things slow down instead of faster.

That only applies if `purge_nonempty_nulls()` is called at all. If `may_have_nonempty_nulls()` returns false, `purge_nonempty_nulls()` shouldn't be called at all.

Reply:
In fact, I don't think `purge_nonempty_nulls()` should be called if `has_nonempty_nulls()` returns `false`. That's a more expensive check, but not as expensive as needlessly reseating a column.

I should peruse the implementation of `purge_nonempty_nulls()`.

Reply:
Please check it out: https://nvidia.slack.com/archives/CCWT6NJ0K/p1673652348546619. The issue was due to using `purge_nonempty_nulls` with that false positive check, and was mitigated by switching to use `has_nonempty_nulls` check.

Reply:
Ok. That's what I was talking about with `has_nonempty_nulls()`.

I think I see your point. If this is to decide whether or not to call `purge_nonempty_nulls()`, we should use `has_nonempty_nulls()` (which internally uses may_have_nonempty_nulls()`). Maybe there's little use for `may_have_nonempty_nulls()` in the JNI API.